From 27a082c8307c1e84b725711d4d1f61a8b57d74e6 Mon Sep 17 00:00:00 2001
From: Francisco Boni <boboniboni@gmail.com>
Date: Sun, 19 Dec 2021 03:38:58 -0300
Subject: [PATCH 2/2] STATIC build

---
 src/CMakeLists.txt | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index dd2fd2b6..b30f533d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -227,8 +227,8 @@ ki18n_wrap_ui(konsoleprivate_SRCS
 # add the resource files for the ui files
 qt5_add_resources( konsoleprivate_SRCS ../desktop/konsole.qrc)
 
-add_library(konsoleprivate ${konsoleprivate_SRCS})
-generate_export_header(konsoleprivate BASE_NAME konsoleprivate)
+add_library(konsoleprivate STATIC ${konsoleprivate_SRCS})
+#generate_export_header(konsoleprivate BASE_NAME konsoleprivate)
 
 target_link_libraries(konsoleprivate
     PUBLIC
@@ -243,12 +243,12 @@ target_link_libraries(konsoleprivate
     ${konsole_LIBS}
 )
 
-set_target_properties(konsoleprivate PROPERTIES
-    VERSION ${KONSOLEPRIVATE_VERSION}
-    SOVERSION ${KONSOLEPRIVATE_SOVERSION}
-)
+#set_target_properties(konsoleprivate PROPERTIES
+    #VERSION ${KONSOLEPRIVATE_VERSION}
+    #SOVERSION ${KONSOLEPRIVATE_SOVERSION}
+#)
 
-install(TARGETS konsoleprivate ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
+#install(TARGETS konsoleprivate ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
 
 add_library(konsoleapp SHARED Application.cpp
    MainWindow.cpp
-- 
2.34.1

