From 13fe6996548dd1456cee24c0e036b8cdcb09f6b4 Mon Sep 17 00:00:00 2001
From: Francisco Boni <boboniboni@gmail.com>
Date: Sun, 19 Dec 2021 03:38:58 -0300
Subject: [PATCH 2/2] STATIC build

---
 src/CMakeLists.txt | 34 ++++++++++++++++++++++------------
 1 file changed, 22 insertions(+), 12 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index dd2fd2b6..ddcacfe6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -227,9 +227,19 @@ ki18n_wrap_ui(konsoleprivate_SRCS
 # add the resource files for the ui files
 qt5_add_resources( konsoleprivate_SRCS ../desktop/konsole.qrc)
 
-add_library(konsoleprivate ${konsoleprivate_SRCS})
+add_library(konsoleprivate STATIC ${konsoleprivate_SRCS})
 generate_export_header(konsoleprivate BASE_NAME konsoleprivate)
 
+#set_property(TARGET konsoleprivate_core PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+#set_property(TARGET konsolecolorscheme PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+#set_property(TARGET keyboardtranslator PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+#set_property(TARGET konsolehelpers PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+#set_property(TARGET konsolesession PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+#set_property(TARGET konsoleprofile PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+#set_property(TARGET konsolecharacters PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+#set_property(TARGET konsoledecoders PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+#set_property(TARGET konsoleprivate PROPERTY VISIBILITY_INLINES_HIDDEN OFF)
+
 target_link_libraries(konsoleprivate
     PUBLIC
     konsoleprivate_core
@@ -243,14 +253,14 @@ target_link_libraries(konsoleprivate
     ${konsole_LIBS}
 )
 
-set_target_properties(konsoleprivate PROPERTIES
-    VERSION ${KONSOLEPRIVATE_VERSION}
-    SOVERSION ${KONSOLEPRIVATE_SOVERSION}
-)
+#set_target_properties(konsoleprivate PROPERTIES
+    #VERSION ${KONSOLEPRIVATE_VERSION}
+    #SOVERSION ${KONSOLEPRIVATE_SOVERSION}
+#)
 
-install(TARGETS konsoleprivate ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
+#install(TARGETS konsoleprivate ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
 
-add_library(konsoleapp SHARED Application.cpp
+add_library(konsoleapp STATIC Application.cpp
    MainWindow.cpp
    settings/ConfigurationDialog.cpp
    settings/ConfigDialogButtonGroupManager.cpp
@@ -275,12 +285,12 @@ target_link_libraries(konsoleapp
   KF5::Crash
 )
 
-set_target_properties(konsoleapp PROPERTIES
-    VERSION ${KONSOLEPRIVATE_VERSION}
-    SOVERSION ${KONSOLEPRIVATE_SOVERSION}
-)
+#set_target_properties(konsoleapp PROPERTIES
+    #VERSION ${KONSOLEPRIVATE_VERSION}
+    #SOVERSION ${KONSOLEPRIVATE_SOVERSION}
+#)
 
-install(TARGETS konsoleapp ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
+#install(TARGETS konsoleapp ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} LIBRARY NAMELINK_SKIP)
 
 set(konsole_SRCS
    main.cpp
-- 
2.34.1

